s3-import-commons/src/test/java/no/unit/nva/publication/s3imports/FileEntriesEventEmitterTest.java

```diff
@@ -134,17 +135,24 @@ public void shouldAcceptEventsWithTopicEqualToFilenameEmissionTopic() {
         InputStream validEvent = createRequestEventForFile(importRequestForExistingFile);
         assertThat(importRequestForExistingFile.getTopic(), is(equalTo(FILENAME_EMISSION_EVENT_TOPIC)));
         assertDoesNotThrow(() -> handler.handleRequest(validEvent, outputStream, CONTEXT));
-        
-        ImportRequest invalidRequest = importRequestForExistingFile.withTopic(randomString());
+    
+        EventReference invalidRequest = changeTopic(importRequestForExistingFile, randomString());
```

Comment:
`withTopic` was a "clone" method that does not exist in EventReference and it seems that was used only in tests. It has been replaced by a method in the test class.